sdk/mobile-center/src/main/java/com/microsoft/azure/mobile/channel/DefaultChannel.java

```diff
@@ -275,33 +297,41 @@ private void suspend(boolean deleteLogs, Exception exception) {
         }
         if (deleteLogs) {
             for (GroupState groupState : mGroupStates.values()) {
-                handleFailureCallback(groupState);
+                deleteLogsOnSuspended(groupState);
             }
         } else {
             mPersistence.clearPendingLogState();
         }
     }
 
-    private void handleFailureCallback(final GroupState groupState) {
+    private void deleteLogsOnSuspended(final GroupState groupState) {
         final List<Log> logs = new ArrayList<>();
+        final int currentState = mStateChangeCounter;
         mPersistence.getLogs(groupState.mName, CLEAR_BATCH_SIZE, logs, new AbstractDatabasePersistenceAsyncCallback() {
+
             @Override
             public void onSuccess(Object result) {
-                if (logs.size() > 0 && groupState.mListener != null) {
-                    for (Log log : logs) {
-                        groupState.mListener.onBeforeSending(log);
-                        groupState.mListener.onFailure(log, new CancellationException());
-                    }
-                }
-                if (logs.size() >= CLEAR_BATCH_SIZE && groupState.mListener != null) {
-                    handleFailureCallback(groupState);
-                } else {
-                    mPersistence.deleteLogs(groupState.mName);
-                }
+                deleteLogsOnSuspended(groupState, currentState, logs);
             }
         });
     }
 
+    private synchronized void deleteLogsOnSuspended(GroupState groupState, int currentState, List<Log> logs) {
+        if (checkStateDidNotChange(groupState, currentState)) {
+            if (logs.size() > 0 && groupState.mListener != null) {
+                for (Log log : logs) {
+                    groupState.mListener.onBeforeSending(log);
+                    groupState.mListener.onFailure(log, new CancellationException());
+                }
+            }
+            if (logs.size() >= CLEAR_BATCH_SIZE && groupState.mListener != null) {
+                deleteLogsOnSuspended(groupState);
+            } else {
+                mPersistence.deleteLogs(groupState.mName);
```

Comment:
If state gets changed, it will stop deleting logs for previous calls. Seems not right. Just `checkStateDidNotChange` out of the method is enough I think.